src/vm/util.cpp

```diff
@@ -1796,42 +1393,27 @@ size_t GetCacheSizePerLogicalCpu(BOOL bTrueSize)
     STATIC_CONTRACT_NOTHROW;
     STATIC_CONTRACT_GC_NOTRIGGER;
 
-    static size_t maxSize;
-    static size_t maxTrueSize;
+    static volatile size_t s_maxSize;
+    static volatile size_t s_maxTrueSize;
 
-    if (maxSize)
-    {
-        // maxSize and maxTrueSize cached
-        if (bTrueSize)
-        {
-            return maxTrueSize;
-        }
-        else
-        {
-            return maxSize;
-        }
-    }
+    size_t size = bTrueSize ? s_maxTrueSize : s_maxSize;
+    if (size != 0)
+        return size;
+
+    size_t maxSize, maxTrueSize;
 
-    // For x86, always get from cpuid.
-#if !defined (_TARGET_X86_)
     maxSize = maxTrueSize = GetLogicalProcessorCacheSizeFromOS() ; // Returns the size of the highest level processor cache
-#endif
 
-#if defined (_TARGET_X86) || defined(_TARGET_AMD64_)
-    if (maxSize == 0)
```

Comment:
And for x64, the `GetCacheSizeFromCpuId` was never used either because of the OS API returns non-zero on all OSes we care about.